src/shell/xdg/mod.rs

```diff
@@ -60,6 +61,49 @@ impl XdgShellState {
 
         Ok(XdgShellSurface { xdg_surface, surface })
     }
+
+    /// Creates an [`XdgPositioner`].
+    pub fn create_positioner(&self) -> Result<XdgPositioner, GlobalError> {
+        let wm_base = self.xdg_wm_base()?;
+        wm_base
+            .send_constructor(xdg_wm_base::Request::CreatePositioner {}, Arc::new(PositionerData))
+            .map(XdgPositioner)
+            .map_err(From::from)
+    }
+}
+
+/// A trivial wrapper for an [`xdg_positioner::XdgPositioner`].
+///
+/// This wrapper calls [`destroy`][xdg_positioner::XdgPositioner::destroy] on the contained
+/// positioner when it is dropped.
+#[derive(Debug)]
+pub struct XdgPositioner(xdg_positioner::XdgPositioner);
+
+impl std::ops::Deref for XdgPositioner {
```

Comment:
This is abusing Deref, this isn't a smart pointer.

Reply:
The same could be said of `std::lazy::Lazy` or `Cow::Owned`.  Requiring another function to access the contained value just clutters up user code for no particular benefit.

Reply:
Patch removed, will be part of #262 (modified to use that trait).